packages/apps/plugins/plugin-explorer/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useResizeDetector } from 'react-resize-detector';
+
+import { type Space } from '@dxos/client/echo';
+import { createSvgContext, SVG, SVGContextProvider } from '@dxos/gem-core';
+
+import { HierarchicalEdgeBundling, RadialTree, TidyTree } from './layout';
+import { mapGraphToTreeData, type TreeNode } from './types';
+import { SpaceGraphModel } from '../Graph';
+
+// TODO(burdon): Create dge bundling graph using d3.hierarchy.
+// https://observablehq.com/@d3/hierarchical-edge-bundling?intent=fork
+
+type Renderer = (svg: SVGSVGElement, data: any, options: any) => void;
+
+export type LayoutType = 'tidy' | 'radial' | 'edge';
+
+// TODO(burdon): Normalize API and styling.
+
+export type TreeLayoutSlots = {
+  node?: string;
```

Comment:
If this value is passed to `mx`, the proper type for this is `ClassNameValue`, which can be an array, which is also the reason the prop name should be `classNames` with an ‘s’ (or we should discuss renaming all such props).

Reply:
Would slots become `classNameses`? (Only joking -- yes, point taken).